kylin-spark-project/kylin-spark-engine/src/main/scala/org/apache/kylin/engine/spark/job/CuboidAggregator.scala

```diff
@@ -98,9 +98,22 @@ object CuboidAggregator {
             new Column(cdAggregate.toAggregateExpression()).as(id.toString)
           }
         case "TOP_N" =>
-          val schema: StructType = constructTopNSchema(measure.pra)
-          val udfName = UdfManager.register(measure.returnType.toKylinDataType, measure.expression, schema, !reuseLayout)
-          callUDF(udfName, columns: _*).as(id.toString)
+          val schema = StructType(measure.pra.map { col =>
```

Comment:
It's better add a comment here, describe that use new 'TopN AggregateFunction' to implement this function.

Reply:
OK.